sdl2promela/promela/generator.py

```diff
@@ -0,0 +1,182 @@
+from multipledispatch import dispatch
+from typing import List, Set, Text, Tuple, Type, TextIO
+
+from . import model
+
+INDENT = "  "
+
+class Context:
+    stream : TextIO
+    indents : List[str]
+    pending_indent : bool
+
+    def __init__(self, stream : TextIO):
+        self.stream = stream
+        self.indents = []
+        self.pending_indent = False
+
+    def push_indent(self, indent : str):
+        self.indents.append(indent)
+
+    def pop_indent(self):
+        self.indents.pop()
+
+    def _get_indent(self):
+        return "".join((self.indents))
+
+    def output(self, data : str):
+        if self.pending_indent:
+            print(self._get_indent(), file=self.stream, end='')
+            self.pending_indent = False
+        lines = data.splitlines(keepends=True)
+        line_count = len(lines)
+        for i in range(0, line_count):
```

Comment:
`for line in lines`

`self.pending_indent = lines[-1].endswith("\n") if len(lines) > 0 else False`

Reply:
This handles the pending_intent, does not handle the immediate indent. I think it is simpler now

Reply:
You trade line complexity for the number of lines, and the cyclomatic complexity hides within the lines, which is not a great thing....

Reply:
OK, removing the redundanty check for > 0 and iterating over lines[:-1] should do the trick....

Reply:
but intent of the author is better visible IMHO. Everything hides complexity ;) (like `splitlines` 3 lines above ;) )